libraries/lib-registries/ClientData.h

```diff
@@ -410,6 +411,81 @@ class Site
       }
    }
 
+   //! Invoke function on corresponding pairs of ClientData objects that have
+   //! been created in @c this and in another Site with the same factories
+   /*!
+    Beware that the sequence of visitation is not specified.
+    @tparam Function takes two pointers to ClientData, return value is ignored
+    @param other supplies the objects to the second argument of function
+    @param function of type @b Function may assume the precondition, that the
+    arguments are not both null, and if create is true, then neither argument is
+    null.  When neither is null, also the objects have come from the same
+    factory.
+    @param create whether to create objects at vacant slots that correspond to
+    non-vacant slots.  (Never create where there are corresponding nulls.)
+    */
+   template<typename Function>
+   void ForCorresponding(Site &other, const Function &function,
```

Comment:
It's very hard to understand what this method is for doing a commit-by-commit review.